File: cpp_tutorial/10_Heaps/exercises/kth_largest.cpp
#include <iostream>
#include <vector>
#include <queue>

// TODO: Find the K-th largest element in an array
// Example: [3, 2, 1, 5, 6, 4], k = 2 -> Output: 5 (6 is 1st largest, 5 is 2nd)
int findKthLargest(std::vector<int>& nums, int k) {
    // Hint: Use a Min-Heap (priority_queue with std::greater) of size k.
    // Keep adding elements. If size > k, pop the smallest.
    // The top of the heap will eventually hold the K-th largest.
    
    // std::priority_queue<int, std::vector<int>, std::greater<int>> minHeap;

    return 0; // Replace
}

int main() {
    std::vector<int> nums = {3, 2, 1, 5, 6, 4};
    int k = 2;

    std::cout << k << "nd largest element is: " << findKthLargest(nums, k) << std::endl;

    return 0;
}

File: Generated_Tutorials/java_tutorial/09_Heaps/exercises/KthLargest.java
import java.util.PriorityQueue;

public class KthLargest {

    // TODO: Find the K-th largest element in an array
    public static int findKthLargest(int[] nums, int k) {
        // Hint: Use a Min-Heap of size k.
        // If heap size > k, poll() the smallest element.
        // The remaining head will be the k-th largest.
        
        // PriorityQueue<Integer> minHeap = new PriorityQueue<>();
        
        return 0; // Replace
    }

    public static void main(String[] args) {
        int[] nums = {3, 2, 1, 5, 6, 4};
        int k = 2;
        System.out.println(k + "nd Largest: " + findKthLargest(nums, k));
        // Expected: 5
    }
}
